src/Liip/RMT/Prerequisite/ComposerSecurityCheck.php

```diff
@@ -39,14 +39,19 @@ public function execute()
 
         Context::get('output')->writeln('<comment>running composer security check</comment>');
 
-        // run the actual security check
-        $checker = new SecurityChecker();
-        $alerts = $checker->check('composer.lock');
+        // Run the actual security check
+        $process = new Process(['local-php-security-checker', '--format', 'json']);
+        $process->run();
 
-        // exit succesfull if everything is fine
-        if (count($alerts) == 0) {
-            $this->confirmSuccess();
+        $alerts = json_decode($process->getOutput(), true);
+        if (count($alerts) === 0) {
+            if (! $process->isSuccessful()) {
+                Context::get('output')->writeln('<error>Error while trying to execute `local-php-security-checker` command. Are you sure the binary is installed globally in your system?</error>');
+                return;
+            }
```

Comment:
Imho, this `if` section must be check outside of the `if (count($alerts) === 0) {` because you can get a non-successful command but an empty array

Reply:
Unfortunately, if there are some "alerts" (the command returns a list of vulnerabilities) it also returns a return code of 1 which is "unsuccessful".

Moreover, if the binary doesn't exist we also get an empty array so I guessed that was the best way of handling it

Reply:
Ok, I misunderstood your intention. All good then 👍